feat: func env vars and secrets docs (#3328)
aws-amplify/docs: docs/cli/function/env-vars.md

---

> title: Custom environment variables

Reviewer: ```suggestion
title: Environment variables
```

> description: Configure Lambda runtime environment variables

Reviewer: ```suggestion
description: Configure environment variables for AWS Lambda functions
```

---

> Amplify CLI allows you to create custom Lambda environment variables that will be available to your Lambda during runtime. Each Amplify environment can have a different value for an environment variable allowing use cases such as switching between dev and prod URLs depending on the environment.

Reviewer: ```suggestion
Amplify CLI allows you to configure environment variables for your Lambda functions. Each Amplify environment can have a different value for an environment variable. For example, this allows you to switch between dev and prod URLs depending on the environment.
```

> > Environment variables should NOT be used for storing sensitive configuration values such as database passwords, API keys or access tokens. To access these types of values from your function, see [function secrets configuration](~/cli/function/secrets.md).

Reviewer: ```suggestion
> Environment variables should NOT be used for storing sensitive configuration values such as database passwords, API keys or access tokens. Use [function secrets configuration](~/cli/function/secrets.md) instead!
```

## Configuring environment variables

> To add a new function with custom environment variables, run `amplify add function`, select `yes` to the advanced settings prompt and select `yes` to the environment variables configuration prompt. From there, you will be able to specify a key and value for the environment variable.

Reviewer: ```suggestion
To configure a new function with environment variables, run `amplify add function`, select `yes` to the advanced settings prompt and select `yes` to the environment variables configuration prompt. From there, you will be able to specify a key and value for the environment variable.
```

Reviewer: also add the code flow maybe?

> To add custom environment variables to an existing function, run `amplify update function`, and select `Environment variables configuration`. You can then add, update and remove environment variables.

Reviewer: ```suggestion
To configure environment variables for an existing function, run `amplify update function`, and select `Environment variables configuration`. You can then add, update, or remove environment variables.
```

Reviewer: add code flow, maybe?

## Multi-environment flows

> When creating a new Amplify environment using `amplify env add`, Amplify CLI will detect if some functions have environment variables. A prompt will appear asking if you want to apply all environment variable values to the new environment or make edits. If you choose to apply the existing values, you will still be able to make edits anytime using `amplify update function`.

Reviewer: ```suggestion
When creating a new Amplify environment using `amplify env add`, Amplify CLI asks if you want to apply all environment variable values to the new environment or modify them. If you choose to apply the existing values, you can still make edits anytime by running `amplify update function`.
```

> When creating a new Amplify environment using `amplify env add --yes`, Amplify CLI will apply all environment variable values from the current environment to the new environment by default.

Reviewer: ```suggestion
When creating a new Amplify environment using `amplify env add --yes`, Amplify CLI will apply all environment variable values from the current environment to the new environment.
```
